internal/fs/fs.go

```diff
@@ -2154,10 +2154,16 @@ func (fs *fileSystem) OpenDir(
 
 	// Enables kernel list-cache in case of non-zero kernelListCacheTTL.
 	if fs.kernelListCacheTTL > 0 {
+		// Following lock ordering rules: first taking dirInode.RLock() and then fs.lock().
+		fs.mu.Unlock()
```

Comment:
we can unlock fs after adding directory handle (line 2152) and remove the defer lock (line 2141)